### PR TITLE
Restructure robot_interfaces

### PR DESCRIPTION
**Pull request by luator**
# Summary
Adapt to changes in robot_interfaces (see https://github.com/open-dynamic-robot-initiative/robot_interfaces/pull/1).

# Detailed
- Create instances of data, backend and frontend separately.  Adapt the `create` function to only create the RobotBackend with a Finger robot driver; other classes are directly created.
- Make the `demo_real_finger.py` executable so it can be run with `rosrun blmc_robots demo_real_finger.py` from everywhere.  Also wrap the code in a `main` function.
- Auto-reformat real_finger.hpp

# How I Tested
`rosrun blmc_robots demo_real_finger.py` 

# Do Not Merge Before
https://github.com/open-dynamic-robot-initiative/robot_interfaces/pull/1

### CODE REVIEW

File: include/blmc_robots/real_finger.hpp [hunk 1]
```diff
@@ -1,348 +1,376 @@
...
 #include <robot_interfaces/finger.hpp>
 
-namespace blmc_robots {
-
-class RealFinger
-    : public robot_interfaces::Robot<robot_interfaces::finger::Action,
-                                     robot_interfaces::finger::Observation> {
+namespace blmc_robots
+{
+class RealFingerBackend : public robot_interfaces::RobotDriver<
```
Comment (mwuethri) [inline include/blmc_robots/real_finger.hpp]: shouldn't this class be called RealFingerDriver?

File: include/blmc_robots/real_finger.hpp [hunk 2]
```diff
@@ -1,348 +1,376 @@
...
+robot_interfaces::finger::BackendPtr create_real_finger_backend(
+    const std::string &can_0, const std::string &can_1,
+    robot_interfaces::finger::DataPtr robot_data)
+{
+    std::shared_ptr<
+        robot_interfaces::RobotDriver<robot_interfaces::finger::Action,
+                                      robot_interfaces::finger::Observation>>
+        robot = std::make_shared<RealFingerBackend>(can_0, can_1);
+
+    auto server =
```
Comment (mwuethri) [inline include/blmc_robots/real_finger.hpp]: so i guess this should be called backend now

**Comment by luator** [comment]
Sorry about the reformatting. This is something I am preaching all the time myself...
Renaming is done

**Comment by mwuethri** [comment]
Great, thanks! :)

Il giorno lun 16 set 2019 alle ore 13:38 Felix Widmaier <
notifications@github.com> ha scritto:

**Pull request merged**